modules/tinymce/src/plugins/lists/main/ts/listmodel/ComposeList.ts

```diff
@@ -92,10 +109,28 @@ const writeDeep = (scope: Document, cast: Segment[], entry: Entry): Segment[] =>
 };
 
 const composeList = (scope: Document, entries: Entry[]): Optional<SugarElement<HTMLElement>> => {
-  const cast = Arr.foldl(entries, (cast, entry) => {
-    return entry.depth > cast.length ? writeDeep(scope, cast, entry) : writeShallow(scope, cast, entry);
+  let firstCommentEntryOpt: Optional<Entry> = Optional.none();
+
+  const cast = Arr.foldl(entries, (cast, entry, i) => {
+    if (isEntryList(entry)) {
+      return entry.depth > cast.length ? writeDeep(scope, cast, entry) : writeShallow(scope, cast, entry);
+    } else {
+      if (i === 0 && isEntryComment(entry)) {
+        firstCommentEntryOpt = Optional.some(entry);
```

Comment:
Having a side-effect like this feels very fragile. Is there no other way to do this? Even if it's a bit less efficient in compute-time?
And if there is no better way could a comment perhaps be added, to make it easier to understand what's going on for future developers.

Reply:
I don't understand what this comment is doing at all. It feels like a patch rather than addressing the underlying issue.

Reply:
this is actually not strictly related to the issue but since for this issue we need to manage not-list element like `p`, I thought that it was correct to also fix the case with `comments`.

in particular this part is used to manage the case when the `comment` is the first element so something like this:
```
<ol>
  <!-- a -->
  <li>b</li>
</ol>
```

in particular this exception is needed because otherwise we are not able to generate a list to put the `comment` in

I added a comment

Reply:
That seems like an edge case, not something that's worth adding a mutable tracking variable for